_ml-commons-plugin/api.md

```diff
@@ -0,0 +1,968 @@
+---
+layout: default
+title: API
+has_children: false
+nav_order: 99
+---
+
+# ML Commons API 
+
+---
+
+#### Table of contents
+- TOC
+{:toc}
+
+
+---
+
+The Machine Learning (ML) commons API lets you create, train, and store machine learning algorithms both synchronously and asynchronously.  
```

Comment:
Seems not so accurate. ML commons API will let user train ML algorithms synchronously and asynchronously, then store ML model (not algorithm) in ML model index.

User can create/add new algorithms by following this doc https://github.com/opensearch-project/ml-commons/blob/main/docs/how-to-add-new-function.md. But that needs code changes, not possible though calling REST APIs.

Reply:
BTW, only train API supports asynchronous way, other APIs only support synchronous way. Because train process generally takes more resources and time. We may support asynchronous way in future if user requires.